go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/DataDog/test-infra-definitions
 
-go 1.21
+go 1.21.12
+
+toolchain go1.22.4
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
We might align with `datadog-agent`

```suggestion
go 1.22.0
```